Sources/Sentry/SentryProfiler.mm

```diff
@@ -373,6 +387,25 @@ + (void)stopProfilerForReason:(SentryProfilerTruncationReason)reason
     _gCurrentProfiler = nil;
 }
 
+- (void)startMetricProfiler
```

Comment:
Why do you acquire the lock in the `-useTimerWrapper:` etc. methods but not here?

Reply:
There is a lock acquired in `+[startForSpanID:...]` which calls `-[start]` which calls this method. Tricky since it's so far removed, is there something we can do to make it more explicit that this method assumes exclusive access to the global variables? Rename it with a prefix like `_unsafe`? Maybe there's some nice GCD API for this using semaphores I'm not aware of.

Reply:
Oh so I guess the other half of this is that those `-[use...]` methods can be called from tests to swap in the mock objects. Probably don't even really need the locks for that, but was being overly defensive there.

Reply:
I think you don't need locks for `useTimerWrapper` and such.

> Rename it with a prefix like _unsafe

I think that's unnecessary, as you call `startMetricProfiler` in a lock in `SentryProfiler.start`.